test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -937,124 +937,194 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 				Expect(running).To(Equal(true))
 
-				By("Ensure there is a Running nginx client two pod")
-				running, err = pod.WaitOnReady(clientTwoDeploymentName, nsClientTwo, 3, retryTimeWhenWaitingForPodReady, cfg.Timeout)
+				By("Ensure there is a running frontend-dev pod")
+				running, err = pod.WaitOnReady(frontendDevDeploymentName, nsDev, 3, retryTimeWhenWaitingForPodReady, cfg.Timeout)
 				Expect(err).NotTo(HaveOccurred())
 				Expect(running).To(Equal(true))
 
-				By("Ensure there is a Running nginx server pod")
-				running, err = pod.WaitOnReady(serverDeploymentName, nsServer, 3, retryTimeWhenWaitingForPodReady, cfg.Timeout)
+				By("Ensure there is a running backend pod")
+				running, err = pod.WaitOnReady(backendDeploymentName, nsDev, 3, retryTimeWhenWaitingForPodReady, cfg.Timeout)
 				Expect(err).NotTo(HaveOccurred())
 				Expect(running).To(Equal(true))
 
-				By("Ensuring we have outbound internet access from the nginx client one pods")
-				clientOnePods, err := clientOneDeploy.Pods()
+				By("Ensure there is a running network-policy pod")
+				running, err = pod.WaitOnReady(nwpolicyDeploymentName, nsDev, 3, retryTimeWhenWaitingForPodReady, cfg.Timeout)
 				Expect(err).NotTo(HaveOccurred())
-				Expect(len(clientOnePods)).ToNot(BeZero())
-				for _, clientOnePod := range clientOnePods {
-					var pass bool
-					pass, err = clientOnePod.CheckLinuxOutboundConnection(5*time.Second, cfg.Timeout)
+				Expect(running).To(Equal(true))
+
+				By("Ensuring we have outbound internet access from the frontend-prod pods")
+				frontendProdPods, err := frontendProdDeployment.Pods()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(len(frontendProdPods)).ToNot(BeZero())
+				for _, frontendProdPod := range frontendProdPods {
+					pass, err := frontendProdPod.CheckLinuxOutboundConnection(5*time.Second, cfg.Timeout)
 					Expect(err).NotTo(HaveOccurred())
 					Expect(pass).To(BeTrue())
 				}
 
-				By("Ensuring we have outbound internet access from the nginx client one pods")
-				clientTwoPods, err := clientTwoDeploy.Pods()
+				By("Ensuring we have outbound internet access from the frontend-dev pods")
+				frontendDevPods, err := frontendDevDeployment.Pods()
 				Expect(err).NotTo(HaveOccurred())
-				Expect(len(clientTwoPods)).ToNot(BeZero())
-				for _, clientTwoPod := range clientTwoPods {
-					var pass bool
-					pass, err = clientTwoPod.CheckLinuxOutboundConnection(5*time.Second, cfg.Timeout)
+				Expect(len(frontendDevPods)).ToNot(BeZero())
+				for _, frontendDevPod := range frontendDevPods {
+					pass, err := frontendDevPod.CheckLinuxOutboundConnection(5*time.Second, cfg.Timeout)
 					Expect(err).NotTo(HaveOccurred())
 					Expect(pass).To(BeTrue())
 				}
 
-				By("Ensuring we have outbound internet access from the nginx server pods")
-				serverPods, err := serverDeploy.Pods()
+				By("Ensuring we have outbound internet access from the backend pods")
+				backendPods, err := backendDeployment.Pods()
```

Comment:
There seems to be duplication here. Let's condense them to a method that takes a list of `Pods` as a parameter.

Reply:
Because these pods have different purposes, this way we have more readability/granularity.

Reply:
I respectfully disagree. If we really need that, we could specify a one line comment and and also add a godoc comment to annotate the newly created method.

Reply:
Do we need a method? The method will only be called once.
If we really want to condense them together, I think this should be enough.
```
pods := []pod.Pod{}
pods = append(pods, frontendProdPods...)
pods = append(pods, frontendDevPods...)
pods = append(pods, backendPods...)
pods = append(pods, nwpolicyPods...)

for _, pod := range pods {
	pass, err := pod.CheckLinuxOutboundConnection(5*time.Second, cfg.Timeout)
	Expect(err).NotTo(HaveOccurred())
	Expect(pass).To(BeTrue())
}
```

Reply:
Since we wrap each podList within a `By()` construct, I'd advocate a method.

Reply:
How about this in pod.go?

```
func (l *List) CheckLinuxOutboundConnection(sleep, duration time.Duration) (bool, error) {
...

}

func (l *List) ValidateCurlConnection(uri string, sleep, duration time.Duration) (bool, error) {
...
}

func (l *List) CheckWindowsOutboundConnection(url string, sleep, duration time.Duration) (bool, error) {
...
}


```